src/debug/daccess/request.cpp

```diff
@@ -3849,25 +3859,27 @@ void
 ClrDataAccess::EnumWksGlobalMemoryRegions(CLRDataEnumMemoryFlags flags)
 {
     SUPPORTS_DAC;
-    WKS::gc_heap::ephemeral_heap_segment.EnumMem();
-    WKS::gc_heap::alloc_allocated.EnumMem();
-    WKS::gc_heap::finalize_queue.EnumMem();
-    WKS::generation_table.EnumMem();
-    WKS::gc_heap::oom_info.EnumMem();
 
-    if (WKS::generation_table.IsValid())
+    g_gcDacGlobals->ephemeral_heap_segment.EnumMem();
+    g_gcDacGlobals->alloc_allocated.EnumMem();
+    g_gcDacGlobals->finalize_queue.EnumMem();
+    g_gcDacGlobals->generation_table.EnumMem();
+    g_gcDacGlobals->oom_info.EnumMem();
+
+    if (g_gcDacGlobals->generation_table.IsValid())
     {
             // enumerating the generations from max (which is normally gen2) to max+1 gives you
             // the segment list for all the normal segements plus the large heap segment (max+1)
             // this is the convention in the GC so it is repeated here
-            for (ULONG i = GCHeapUtilities::GetMaxGeneration(); i <= GCHeapUtilities::GetMaxGeneration()+1; i++)
+            for (ULONG i = *g_gcDacGlobals->max_gen; i <= *g_gcDacGlobals->max_gen +1; i++)
```

Comment:
Yes, we should consolidate. Note that NUMBERGENERATIONS is 4 and (max_gen + 1) is 3 so make sure you account for that.